src/main/java/jnr/unixsocket/impl/Common.java

```diff
@@ -121,14 +121,18 @@ int write(ByteBuffer src) throws IOException {
         return n;
     }
 
-    long write(ByteBuffer[] srcs, int offset, int length)
-        throws IOException {
+    long write(ByteBuffer[] srcs, int offset, int length) throws IOException {
 
         long result = 0;
-        int index = 0;
 
-        for (index = offset; index < length; index++) {
-            result += write(srcs[index]);
+        for (int index = offset; index < length; ++index) {
+            ByteBuffer buffer = srcs[index];
+            int remaining = buffer.remaining();
+            int written = write(buffer);
+            result += written;
+            if (written < remaining) {
+                break;
```

Comment:
Is it correct to immediately break? rather than try again at least once?
Consider an impl that can only write a fixed size at a time (say 1024), but it can take many calls each slicing off 1024?
Isn't it better to break only when no progress is made? ie written==0

Reply:
@gregw good point. I have changed the code to reflect your review. Good?